### PR TITLE
feat: better install

### PR DESCRIPTION
**Pull request by lukadev-0**
Improves the install experience and refactors parts of the download and installation code.

The video below demonstrates the new experience. Downloads are artificially slowed down.

https://github.com/user-attachments/assets/5b56643c-c33b-40a5-b7b1-fac67e8e6315

### CODE REVIEW

**Review by daimond113 — CHANGES_REQUESTED** [review]
Thanks for the PR! I noticed a few details I'd like addressed, though.

File: src/cli/install.rs [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+                        Err(e) if e.kind() == std::io::ErrorKind::AlreadyExists => {}
+                        e => e.context("failed to copy bin link file")?,
+                    }
+
+                    #[cfg(not(windows))]
+                    fs::write(
+                        &bin_exec_file,
+                        format!(
+                            r#"#!/bin/sh
+                    exec lune run "$(dirname "$0")/.impl/{alias}.luau" -- "$@""#
```
[the remark this answers is not in the document]
Comment (lukadev-0) [reply src/cli/install.rs]: Fixed.

File: src/cli/install.rs [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+                        .await
+                        .context("failed to create bin link folder")?;
+
+                    let bin_file = impl_folder.join(&alias).with_extension("luau");
+                    fs::write(&bin_file, bin_link_file(&alias))
+                        .await
+                        .context("failed to write bin link file")?;
+
+                    #[cfg(windows)]
+                    match fs::symlink_file(
```
[the remark this answers is not in the document]
Comment (lukadev-0) [reply src/cli/install.rs:125]: I did some basic testing and it works fine, though there may be some edge cases I haven't tested for. Might need some thorough testing in the future.

File: src/cli/reporters.rs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+use colored::Colorize;
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use pesde::reporters::{
+    DownloadProgressReporter, DownloadsReporter, PatchProgressReporter, PatchesReporter,
+};
+
+pub const TICK_CHARS: &str = "⣷⣯⣟⣿⢿⣻⣽⣾";
+
+pub fn root_progress_style() -> ProgressStyle {
+    ProgressStyle::with_template("{prefix:.dim}{msg:>8.green} {spinner} [{elapsed_precise}]")
```
[the remark this answers is not in the document]
Comment (lukadev-0) [reply src/cli/reporters.rs:19]: I don't think it looks good.

[the remark this answers is not in the document]
Comment (lukadev-0) [reply src/cli/reporters.rs]: Oh, you meant the message. Sure I'll try.

File: src/main.rs [hunk 4]
```diff
@@ -146,8 +190,9 @@ async fn run() -> anyhow::Result<()> {
         .add_directive("hyper=info".parse().unwrap())
         .add_directive("h2=info".parse().unwrap());
 
-    let fmt_layer =
-        tracing_subscriber::fmt::layer().with_writer(indicatif_layer.inner().get_stderr_writer());
+    let fmt_layer = tracing_subscriber::fmt::layer()
+        .pretty()
```
Comment (daimond113) [inline src/main.rs:194]:
```suggestion

```

We set it to pretty in in `#[cfg(not(debug_assertions))]`

**Review by daimond113 — CHANGES_REQUESTED** [review]
Thanks for the corrections! There's one small issue left, though.

**Review by daimond113 — APPROVED** [review]
LGTM!

**Pull request merged**